velox/type/Timestamp.h

```diff
@@ -33,9 +33,14 @@ namespace facebook::velox {
 struct Timestamp {
  public:
   enum class Precision : int { kMilliseconds = 3, kNanoseconds = 9 };
+  // Nanos in Timestamp need to be less than 1 second.
+  static constexpr uint64_t kMaxNanos = 999'999'999;
+
   constexpr Timestamp() : seconds_(0), nanos_(0) {}
-  constexpr Timestamp(int64_t seconds, uint64_t nanos)
-      : seconds_(seconds), nanos_(nanos) {}
+  Timestamp(int64_t seconds, uint64_t nanos)
+      : seconds_(seconds), nanos_(nanos) {
+    VELOX_DCHECK_LE(nanos, kMaxNanos);
```

Comment:
Should we also add a check for nanos >= 0 ? CC: @Yuhta 

It would be nice to document how timestamps before epoch are expected to be represented. E.g. seconds can be negative, but nanosecons should always be >= 0.

Reply:
nanos in Timestamp is unsigned int64, so it will be >= 0.

Reply:
Hi @mbasmanova , I added a description of TIMESTAMP to types.rst. Could you help to check the accuracy of the expression.

Reply:
> Nanos in Timestamp is unsigned int64, so it will be >= 0.

I see... Actually, I ran into some code which passes negative values and these were silently converted to "positive" values. It would be helpful to add a comment to this class to explain what valid values are and how to specify timestamps before the epoch.